### ISSUE TITLE
Minimal Cell UI

### ISSUE THREAD

**Issue by michael** [stencila/stencila#248]
Sketched some ideas about a more minimal cell editor, and how we could get rid of lines from the UI. I took inspiration from the Atom Linter, which marks problematic areas in the text and shows a tooltip when you move your cursor inside that range. Using that approach we could get rid of the jumpiness entirely when errors are displayed. We could also delay the appearance of the tooltip a bit.. so it doesn't get in the way while you are still editing.

Additionally I'm suggesting to put a minimal output status line after the code. E.g. it could say VegaLite... and by clicking on it you could further inspect the value. E.g. displaying the JSON. Or we could point to the documentation of that value type.

![image](https://cloud.githubusercontent.com/assets/2931/24276578/87def6da-109c-11e7-8807-9aae99d05aeb.png)

**Comment by oliver----** [comment]
Just want to mention that the 'jumpy' error displays come from runtime errors, which are related to the external source code.

**Comment by michael** [comment]
Yeah, I think we should display both error types in the same way though. To the user we just communicate, you have an error/warning you want to investigate. And bringing it up as an overlay only when the cursor is in the code seems like a nice approach for me. We'd need to try out. I'll follow up with some more sketches and get the UI in place, if no objections. I can just show the dot for now and hovering brings up the error message just as now, but having the selection-triggered overlay would be nicer in the long run i think.

**Comment by michael** [comment]
Think we can hide the visibility settings within in the ellipsis menu.

![image](https://cloud.githubusercontent.com/assets/2931/24280894/4d25621e-10b9-11e7-83ce-1ff1a94592f2.png)

Cell outputs of core types:

![image](https://cloud.githubusercontent.com/assets/2931/24280737/e2ccd7d6-10b7-11e7-95cf-c0b85a5a4353.png)

Whenever there is an error and we had a previous result, we just grey it out. I think it's always useful to see the old result (e.g. in a graph) and then once the expression compiles again, refreshes the display. Then also we don't get jumpiness. We may want to put a "Clear output" option into the ellipsis menu.

![image](https://cloud.githubusercontent.com/assets/2931/24280868/322eab50-10b9-11e7-8c80-deafeb6bae1d.png)

Call Cells (with language indicator)

![image](https://cloud.githubusercontent.com/assets/2931/24280941/a8c4116a-10b9-11e7-8cb1-a05eac4f2c7f.png)


Ellipsis menu for a call cell:

![image](https://cloud.githubusercontent.com/assets/2931/24280924/838ff4c2-10b9-11e7-92fe-6daa91ea8a1c.png)
